### ISSUE TITLE
Validation renderer gets an instance of a class, not the html element

### ISSUE THREAD

**Issue by MaximBalaganskiy** [aurelia/validatejs#63]
https://gist.run/?id=66e783f15c68f439adfee440d55438ca

As you can see I validate the md-value on a custom element. The renderer `md-validation-renderer.js` receives a parameter `node` which is an instance of an `md-input` class defined by the bridge. Luckily, it has an `element` field which I can use. But I cannot assume that all custom elements will expose an HTML element.

In a nutshell, I don't particularly like this code in the renderer
```
    if (node.element)
        this.renderErrorsOnElement(node.element, relevantErrors);
    else if (node.classList)
        this.renderErrorsOnElement(node, relevantErrors);
```

**Comment by ggrimbert** [comment]
Solved by this : 

```
<div ref="parentElement">
<input XXXXX>
</div>
```

Doing this, the node passed gets a parentElement even if this is not an html component.

**Comment by MaximBalaganskiy** [comment]
Should this be added to an element template?

**Comment by ggrimbert** [comment]
This doesn't work with template, I cannot remember why...
But you can add the div inside the template like this : 
```
<template>
<div ref="parentElement">
[...]
</div>
</template>

**Comment by MaximBalaganskiy** [comment]
That's fine for my own controls. What about 3rd party controls like Materialize which might not have element or parentElement exposed?

**Comment by ggrimbert** [comment]
I haven't integrated this, so I can't help you

**Comment by PWKad** [comment]
@MaximBalaganskiy can you summarize the changes you are proposing or put together a PR?

**Comment by MaximBalaganskiy** [comment]
@PWKad I'm not that familiar with the framework to submit a PR, I'm afraid.

I think, renderer must have access to an HTML representing the custom element. Are there any framework service classes which can take a class instance and return a UI tree? If yes it can be injected into a renderer.

**Comment by JeroenVinke** [comment]
I have the same issue with the validate binding behavior on a custom element / attribute. Used the following code to reproduce:

```html
<template>
  <input type="text" value.bind="mdValue"/ />
</template>
```

```js
import {bindable} from 'aurelia-framework';
export class MdInput {
  @bindable mdValue;
}
```

```html
<div>
  <label class="form-group">
    <strong>First Name</strong>
    <!-- <input class="form-control" value.bind="model.firstName & validate" /> -->
    <md-input md-value.two-way="model.firstName & validate"></md-input>
  </label>
</div>
```

An assumtion is made [here](https://github.com/aurelia/validatejs/blob/master/src/validate-binding-behavior.js#L22) that `binding.target` is the HTML element, which is true for `<input value.bind="model.firstName & validate"/>` but when a custom element is used `binding.target` is the instance of the MdInput viewmodel

**Comment by JeroenVinke** [comment]
if anyone is interested, I made a workaround: https://github.com/JeroenVinke/validatejs/commit/7876e9a4ec124cd8e510e4ba69fa7c452f1473c1

It is used like this: `<md-input ref="_firstName" md-value.two-way="model.firstName & validate:_firstName"></md-input>`

`jspm install aurelia-validatejs=github:JeroenVinke/validatejs@master`

**Comment by jdanyow** [comment]
this is fixed in the latest build
